### ISSUE TITLE
RDMA support for g6e nodes

### ISSUE THREAD

**Issue by Abhishek8394** [aws/aws-ofi-nccl#547]
Hi, I've been trying out nccl tests on g6e nodes and it maxes out at 3GBps. I see that gpudirect rdma is not enabled on g6e nodes when trying to use aws nccl plugin, with efa enabled on 4 network interfaces. Nvidia states that those gpus do support gpudirect rdma.

Is it a matter of having a topology file or something more needed?

Would love to contribute support if given some pointers.

**Comment by bwbarrett** [comment]
Sorry for the lack of a reply on this issue; not sure how this fell through the cracks.

Although the L40S GPUs support GPUDirect RDMA, the G6e hardware does not support GPUDirect RDMA.  The system was designed primarily for single node inference, and has not been a focus for the EFA-related teams.

**Comment by bwbarrett** [comment]
No, at present (Aug '25), the only instance types in EC2 that support GPUDirect RDMA are the P-series starting with p4d.

**Comment by alokprasad** [comment]
@bwbarrett Thanks for clarification on GPUDirect Part, but on the RDMA Write part i see that Nitro v3 dosent have that support
and NCCL uses RDMA Write , so how does NCCL Work in that case?

**Comment by bwbarrett** [comment]
Nothing in NCCL assumes RDMA Write is available.  The NCCL interface to the underlying network is a send/recv interface.

**Comment by alokprasad** [comment]
@bwbarrett 
while running all_reduce_perf with Nitro V3 i see following abort message.
```
FI_EFA_USE_DEVICE_RDMA=1 was set by user, but EFA device has no rdma-read capability.  Application will abort().
```

**Comment by alokprasad** [comment]
it's g4dn.8xlarge
